src/middleware/requireAuthentication.ts

```diff
@@ -8,7 +8,15 @@ const requireAuthentication = (
 	next: NextFunction,
 ) => {
 	if (req.auth === undefined) {
-		next(new UnauthorizedError('No authorization token was found'));
+		next(new UnauthorizedError('No authorization token was found.'));
+		return;
+	}
+	if (req.auth?.sub === undefined || req.auth.sub === '') {
```

Comment:
Is the authorization subject being empty string really it _lacking_ the subject, or a separate issue?

Reply:
I'd say an empty string is lacking a subject.  Or rather, with this line we are saying that in our system we aren't considering empty string to be a valid subject.

Reply:
After a few moments thought I think that's sufficient, the subject is meant to be a self-describing string correct? I.e `system` or `client` . If that's the case, then an empty string is 'lacking', as the only validity check is if it's a readable string that could communicate it's purpose.